livre/core/render/TransferFunction1D.cpp

```diff
@@ -40,149 +33,85 @@ namespace
 enum SamplePointsType
 {
     UINT8,
-    FLOAT,
-    DEFAULT = FLOAT
+    FLOAT
 };
+}
 
-Floats create2ColorTF( const uint32_t surfaceColorPos,
-                            const float surfaceColorAlpha,
-                            const float solidColorAlpha, const size_t size,
-                            const Vector3f& surfaceColorRGB,
-                            const Vector3f& solidColorRGB )
+TransferFunction1D::TransferFunction1D()
 {
-    Vector3f finalColorRGB;
-    float alpha;
-    Floats transferFunction;
-
-    const Vector3f colorDiff = solidColorRGB - surfaceColorRGB;
-    const float alphaDiff = solidColorAlpha - surfaceColorAlpha;
-
-    for( uint32_t i = 0; i < size; ++i )
+    const Vector3f color1( 0.0f, 1.0f, 1.0f );
+    const Vector3f color2( 1.0f, 0.0f, 1.0f );
+    const float alpha1 = 0.39f;
+    const float alpha2 = 0.03f;
+    const float density1 = 127.f;
+    const float density2 = 255.f;
+    const Vector3f colorDiff = color2 - color1;
+    const float alphaDiff = alpha2 - alpha1;
+
+    uint8_t* data = getLut();
+
+    for( uint32_t i = 0; i < getLutSize(); i += NCHANNELS )
     {
-        if( i == 0 )
+        Vector4f rgba;
+        if( 0 < i && i <= density1 )
         {
-            finalColorRGB = Vector3f(0.0f);
-            alpha = 0;
+            rgba = color1;
+            rgba[3] = alpha1 * ( float(i) / density1 );
         }
-        else if( ( 0 < i ) && ( i<= surfaceColorPos ) )
+        else if( i > 0 )
         {
-            finalColorRGB = surfaceColorRGB;
-            alpha = surfaceColorAlpha * ( float(i) / (float)surfaceColorPos );
-        }
-        else
-        {
-            finalColorRGB = ( (float)( i - surfaceColorPos + 1u ) /
-                              (float)( DEFAULT_COLOR2_DENSITY -
-                                       surfaceColorPos + 1u ) ) * colorDiff +
-                             surfaceColorRGB;
-            alpha = ( (float)( i - surfaceColorPos + 1u ) /
-                    (float)( DEFAULT_COLOR2_DENSITY - surfaceColorPos + 1u ) ) *
-                    alphaDiff + surfaceColorAlpha;
+            rgba = ( float( i ) - density1 + 1.f ) /
+                   ( density2 - density1 + 1.f ) * colorDiff + color1;
+            rgba[3] = ( float( i ) - density1 + 1.f ) /
+                      ( density2 - density1 + 1.f ) * alphaDiff + alpha1;
         }
+        // else i == 0 : rgba = 0;
 
-        transferFunction.push_back( finalColorRGB[0] );
-        transferFunction.push_back( finalColorRGB[1] );
-        transferFunction.push_back( finalColorRGB[2] );
-        transferFunction.push_back( alpha );
+        data[ i + 0 ] = rgba[0] * std::numeric_limits< uint8_t >::max();
```

Comment:
You mean in this file or in the entire project ? What do you mean by "as float" ? 255.f ?